src/features/payment-agreements/hooks/mutations/use-create-payment-agreement-mutation.ts

```diff
@@ -27,6 +27,8 @@ export function useCreatePaymentAgreementMutation() {
     onSuccess(sessionId) {
       if(!sessionId) return Promise.reject();
 
+      client.resetQueries({queryKey: accountPaymentAgreementsCreatedQueryKey(address), exact: true})
```

Comment:
Success callbacks (client.resetQueries) should be added to `getCallback` function and not placed directly in `onSuccess`. 

`onSuccess` is called as soon as the transaction is signed, not when it finishes executing on the blockchain. We need to run our callbacks IF and WHEN the transaction is successfuly executed.